### DISCUSSION TITLE
Polydata.n_verts gives zero while mesh definetely has vertices

### DISCUSSION THREAD

**Discussion by MarioAuditore** [pyvista/pyvista#4592] [Q&A] [answered]
So, I have human mesh: [https://drive.google.com/file/d/1j0phtyG7noq6T6hLMQT8YBASfjBLdTFZ/view?usp=drive_link](url)

It is absolutely normal file, which can be viewed in blender etc. Pymeshlab is able to read the mesh and it shows that it has 24997 vertices. However, in pyvista's polydata I get same amount of faces, but zero vertices...

```python
mesh_path = 'slastnikov.obj'

reader = pv.get_reader(mesh_path)
mesh = reader.read()
mesh = mesh.rotate_x(90, inplace=False).rotate_z(180, inplace=False)

print(mesh.n_verts, mesh.n_faces)
```
result: 0 50000

It seems insane, because when I plot the mesh, the vertices can be clearly seen
<img width="377" alt="image" src="https://github.com/pyvista/pyvista/assets/57262352/c7f41d79-bf7f-428f-8111-75f1fe4847ba">

Any ideas?

**Comment by darikg** [comment, marked as answer]
I think you want `mesh.n_points`. IIUC a vertex in VTK is a single element cell which references a single point, in the same way a triangle references 3 points. So your mesh has lots of points but no vertices, if that makes sense.

**Reply by banesullivan** [reply]
Exactly correct! I want to add one more bit that `n_verts` is specific to PolyData (though vertex cells can also exist in UnstructuredGrids) and that `n_points` is the API present to get the number of points for *all* mesh data types

**Reply by banesullivan** [reply]
The vtkPolyData is https://vtk.org/doc/nightly/html/classvtkPolyData.html#ac9d21209cb08bf5f2ca4f0f43ffdd533

And note that PolyData has this for Verts, Lines, Polys (faces), and Strips

**Reply by MatthewFlamm** [reply]
After further investigation, `n_verts` is actually the sum of all 0-D cells.  This is almost always a single cell that references one point.  It can also be a single cell that references multiple points, but without topological information that would give it a spatial dimension.  Subtle but could be important for the rare person who is using a mesh with the latter type of cell.
